docs/saml-2-0/mapping.md

```diff
@@ -0,0 +1,91 @@
+# Mapping Data
+
+The SAML SSO functionality includes out-of-the-box functionality for mapping data from your identity provider (IdP) to WordPress-native data.
+
+This mapping can be customized and extended to match the way your IdP stores data.
+
+
+## User details
+
+By default, the following fields are mapped:
+
+* `user_login` field (equivalent to a user slug): `email` SAML attribute
+* `user_email`: `email` SAML attribute
+* `first_name`: `firstName` SAML attribute
+* `last_name`: `lastName` SAML attribute
+
+This can be filtered via the `wpsimplesaml_attribute_mapping` filter, which receives an associative array mapping user properties to SAML attribute name. (Note: only these four fields are supported.)
+
+For example, for Active Directory (AD), you may want to use the [AD claims](https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/technical-reference/the-role-of-claims) instead:
+
+```php
+add_filter( 'wpsimplesaml_attribute_mapping', function() {
+	return [
+		'user_email' => 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress',
+		'first_name' => 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname',
+		'last_name'  => 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname',
+		'user_login' => 'http://schemas.microsoft.com/identity/claims/displayname',
+	];
+} );
+```
+
+### Advanced user details
+
+Aside from the direct mapping of these , you can also filter the final data before it is sent to WordPress, via the `wpsimplesaml_user_data` filter.
+
+This filter receives the full user data being passed to `wp_insert_user` if you need to apply any other customization to this user data. You'll also receive the SAML attributes.
```

Comment:
Would this read easier if split into two sentences? And maybe mention adding metadata:

"This filter receives the full user data being passed to `wp_insert_user`. You can use it to apply any other customization to this user data (including adding meta data). You'll also receive the SAML attributes.